app/api/events.py

```diff
@@ -287,7 +287,7 @@ def after_create_object(self, event, data, view_kwargs):
 class EventDetail(ResourceDetail):
 
     def before_get_object(self, view_kwargs):
-        if view_kwargs.get('identifier'):
+        if view_kwargs.get('event_identifier'):
```

Comment:
Resolved. I am looking for other errors that might have crept in after the `safe_query` addition. 
@shubham-padia Please check for yourself too :smile: